src/main/java/io/cdap/plugin/common/SFTPActionConfig.java

```diff
@@ -43,8 +43,24 @@
   @Macro
   public String userName;
 
+  @Description("Private Key to be used to login to SFTP Server. SSH key must be of RSA type")
+  @Macro
+  @Nullable
+  public String privateKey;
+
+  @Description("Passphrase to be used with private key if passphrase was enabled when key was created. " +
+    "If PrivateKey is selected for Authentication")
+  @Macro
+  @Nullable
+  public String passphrase;
+
+  @Name("Authentication")
+  @Description("Authentication type to be used for connection")
```

Comment:
we shouldnt add a new property that is required. This is a backwards incompatible change.

Reply:
fixed.